File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Messaging/Type/Authentication/AuthenticationDisconnect.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;
using KingstonDynamics.DragonSong.Protocol.Utils;

namespace KingstonDynamics.DragonSong.Protocol.Messaging.Type.Authentication
{
    public class AuthenticationDisconnect : Message
    {
        private const MessageType Type = MessageType.AUTHENTICATION_DISCONNECT;

        public AuthenticationDisconnect(string auditId) : base((int)Type, auditId)
        {
            // Empty
        }

        public AuthenticationDisconnect() : this(Keyinator.GenerateGUID())
        {
            // Empty
        }

        public AuthenticationDisconnect(Readinator reader) : base(reader)
        {
            // Empty
        }
    }
}

File: src/protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/message/type/authentication/AuthenticationDisconnect.java
package com.kingstondynamics.dragonsong.protocol.message.type.authentication;

import com.kingstondynamics.dragonsong.protocol.message.Message;
import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;
import com.kingstondynamics.dragonsong.protocol.utils.Keyinator;
import com.kingstondynamics.dragonsong.protocol.utils.Readinator;

public class AuthenticationDisconnect extends Message {
    
    private final static MessageType type = MessageType.AUTHENTICATION_DISCONNECT;
    
    public AuthenticationDisconnect(String auditId) {
        super(type.getType(), auditId);
    }
    
    public AuthenticationDisconnect() {
        this(Keyinator.generateGUID());
    }
    
    public AuthenticationDisconnect(Readinator reader) {
        super(reader);
    }
}
